### PR TITLE
extract eventLoop out from ServerFactory and Server

### PR DESCRIPTION
**Pull request by kortemik**

### CODE REVIEW

**Comment by kortemik** [comment]
closes #152

File: src/test/java/com/teragrep/rlp_03/readme/ReadmeDeferredTest.java [hunk 1]
```diff
@@ -124,37 +125,39 @@ public void close() {
...
-        Server server;
+        EventLoopFactory eventLoopFactory = new EventLoopFactory();
+        EventLoop eventLoop;
         try {
-            server = serverFactory.create(listenPort);
+            eventLoop = eventLoopFactory.create();
         }
-        catch (IOException ioException) {
-            throw new UncheckedIOException(ioException);
+        catch (IOException e) {
```
Comment (StrongestNumber9) [inline src/test/java/com/teragrep/rlp_03/readme/ReadmeDeferredTest.java:135]: Should be fixed to Assertions

File: src/test/java/com/teragrep/rlp_03/readme/ReadmeTest.java [hunk 2]
```diff
@@ -101,37 +102,39 @@ public FrameDelegate get() {
...
-        Server server;
+        EventLoopFactory eventLoopFactory = new EventLoopFactory();
+        EventLoop eventLoop;
         try {
-            server = serverFactory.create(listenPort);
+            eventLoop = eventLoopFactory.create();
         }
-        catch (IOException ioException) {
-            throw new UncheckedIOException(ioException);
+        catch (IOException e) {
```
Comment (StrongestNumber9) [inline src/test/java/com/teragrep/rlp_03/readme/ReadmeTest.java:112]: Assertions this

File: src/test/java/com/teragrep/rlp_03/readme/ReadmeTest.java [hunk 3]
```diff
@@ -101,37 +102,39 @@ public FrameDelegate get() {
...
-         * Wait for startup, server is available for connections once it finished setup
-         */
         try {
-            server.startup.waitForCompletion();
+            serverFactory.create(listenPort);
             System.out.println("server started at port <" + listenPort + ">");
         }
-        catch (InterruptedException interruptedException) {
-            throw new RuntimeException(interruptedException);
+        catch (IOException ioException) {
```
Comment (StrongestNumber9) [inline src/test/java/com/teragrep/rlp_03/readme/ReadmeTest.java:136]: Assertions this

**Review by StrongestNumber9 — APPROVED** [review]
lgtm

**Pull request merged**